components/http_server/include/http_server.h

```diff
@@ -70,6 +70,69 @@ typedef void* httpd_handle_t;
  */
 typedef enum http_method httpd_method_t;
 
+/* ************** Group: TX/RX ************** */
+/** @name TX / RX
+ * Prototype for HTTPDs low-level send/recv-related functions
+ * @{
+ */
+
+/**
+ * @brief  Prototype for HTTPDs low-level send function
+ * @param[in] hd      : server instance
+ * @param[in] sockfd  : session socket file descriptor
+ * @param[in] buf     : buffer with bytes to send
+ * @param[in] buf_len : data size
+ * @param[in] flags   : flags for the send() function
+ * @return
+ *  - Bytes : The number of bytes sent successfully
+ *  - -VE   : In case of error
+ */
+typedef int (*httpd_send_func_t)(httpd_handle_t hd, int sockfd, const char *buf, size_t buf_len, int flags);
```

Comment:
this can be done, but it seems needlessly convoluted to me. Is there a reason to worry about API changes, if the http_server component is still unreleased in a stable version? (this is why I'm using master instead of the stable release)

while we're discussing it, what was the original purpose of the send / recv override functions? Because if they are used for SSL handling, they become unavailable for anything else.

Reply:
> Is there a reason to worry about API changes, if the http_server component is still unreleased in a stable version?

Your are right. But my only concern is that people might have already started using the API. Still let me discuss this and get back to you.

> while we're discussing it, what was the original purpose of the send / recv override functions?

The purpose was to allow for extension/encryption/encapsulation of HTTP packets as per requirement, just like you are using it for SSL

> Because if they are used for SSL handling, they become unavailable for anything else.

That is alright. They are intended to allow one layer of extension only. Now in case someone requires something fancy with SSL as well as some form of encapsulation then she will have to start implementing on top of http_server.

Reply:
@MightyPork After some discussion we have concluded that changing the API should be alright in this case, like you said